### PR TITLE
tribits_ctest_driver(): Use more robust and better git commands and stronger testing (TRIL-260)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
CC: @fryeguy52 

## Description

TriBITS Snapshot commit:

```
Origin repo remote tracking branch: 'github/master'
Origin repo remote repo URL: 'github = git@github.com:TriBITSPub/TriBITS.git'

At commit:

commit 49d9c814defe2ec0019957233347c87223b04a7e
Author:  Roscoe A. Bartlett <rabartl@sandia.gov>
Date:    Thu Feb 14 13:42:11 2019 -0700
Summary: tribits_ctest_driver(): Use more robust and better git commands and stronger testing (#278, TRIL-260)
```

This brings in the TriBITS commits:

```
49d9c81 "tribits_ctest_driver(): Use more robust and better git commands and stronger testing (#278, TRIL-260)"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Feb 14 13:42:11 2019 -0700 (14 hours ago)

2541677 "Update some README files for examples (#278)"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Tue Feb 19 06:41:51 2019 -0700 (14 hours ago)

5e1677e "Add generic ctest -S driver to submit to CDash"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Thu Feb 14 18:23:30 2019 -0700 (14 hours ago)

64a7ba1 "Add more documentation for <Project>_SHOW_TEST_START_END_DATE_TIME"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Tue Feb 19 11:52:56 2019 -0700 (18 hours ago)

f5ca5b7 "Fix RawAndTribitsHelloWorld/README (#276)"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Tue Feb 12 21:25:33 2019 -0700 (7 days ago)

b4d8233 "Add RawAndTribitsHelloWorld example (#276)"
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Tue Feb 12 20:15:00 2019 -0700 (7 days ago)
```

## Motivation and Context

The main driver for this snapshot it to make the git update commands used in `tribits_ctest_driver()` more robust to avoid situations like described in [TRIL-260](https://sems-atlassian-son.sandia.gov/jira/browse/TRIL-260?focusedCommentId=25911&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-25911) with the ATDM Trilinos builds. 

## How Has This Been Tested?

A bunch of new automated testing, including real git clones, etc.

### CODE REVIEW

File: cmake/tribits/ctest_driver/TribitsGetCTestTestXmlDir.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#
+# cmake -P script to get the CTest testing XML directory
+# <build>/Testing/<buildstarttime> given jus the <bulid> directory path.
```
Comment (mhoemmen) [inline cmake/tribits/ctest_driver/TribitsGetCTestTestXmlDir.cmake]: "just"

**Comment by bartlettroscoe** [comment]
@mhoemmen, thanks for the review & approval!  I will fix in new TriBITS commits and they will come over to Trilinos in a future snapshot.

File: cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst [hunk 2]
```diff
@@ -5871,6 +5873,75 @@ one would perform the following steps:
...
+
+  cmake [options] \
+    -DRawAndTribitsHelloWorld_TRIBITS_DIR=<tribits_dir> \
+     <some_base_dir>/RawAndTribitsHelloWorld
+
+A twist on this use case is for a package that only builds as a TriBITS
+package inside of some larger TriBITS project and not as its own TriBITS CMake
+project.  In this case, some slight changes are needed to this example but the
+basic approach is nearly identical.  One still needs a ``if()`` statement at
+the top of first ``CMakeLists.txt`` file (this time for the package) and the
```
Comment (fryeguy52) [inline cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst]: **an** ``if()`` statement

the top of **the** first

File: cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst [hunk 3]
```diff
@@ -8426,15 +8499,16 @@ These options are described below.
...
+  to be printed as well.  The TriBITS default is ``OFF`` but a TriBITS project
+  can change this default by setting::
 
     SET(${PROJECT_NAME}_SHOW_TEST_START_END_DATE_TIME_DEFAULT ON)
 
   The implementation of this feature currently uses ``EXECUTE_PROCESS(date)``
-  and therefore will only work on Linux/Unix/Mac systems and not Windows
-  systems.
+  and therefore will only work on many (but perhaps not all) Linux/Unix/Mac
+  systems and not Windows systems.
```
Comment (fryeguy52) [inline cmake/tribits/doc/developers_guide/TribitsDevelopersGuide.rst]: and therefore will work on many (but perhaps not all) Linux/Unix/Mac
  systems but not on Windows systems.

File: cmake/tribits/examples/TribitsExampleProject/README [hunk 4]
```diff
@@ -2,50 +2,36 @@
...
-
-You can create another directory for SERIAL_RELEASE or whatever build type you
-want.
+  cd ~/TribitsExampleProject.base/BUILDS/GCC-<X.Y.Z>/BUILDS/SERIAL_DEBUG
+  cp ../../../../TribitsExampleProject/sampleScripts/linux-serial-debug.sh \
+    do-configure
 
-If your system is already to set up to find the MPI compiler wrappers, you con
-just configure with:
+After that set, configure with:
```
Comment (fryeguy52) [inline cmake/tribits/examples/TribitsExampleProject/README]: > After that set, configure with:

is confusing to me.  "After that is set, configure with:" or "After that, configure with:"

File: cmake/tribits/examples/TribitsExampleProjectAddons/README [hunk 5]
```diff
@@ -2,20 +2,25 @@
...
-more in the future to be more complete.
+The project TribitsExampleProjectAddons defines a CMake TriBITS Repository
+designed to provide a simple example to demonstrate how to use TriBITS
+repositories with the TriBITS system to add on additional TriBITS CMake
+packages to an existing TriBITS CMake project.
 
-The project TribitsExampleProjectAddons defines a TriBITS CMake project
-designed to provide a simple example to demonstrate how to use extra
-repositories with the TriBITS system.
+This is TriBITS repository is designed to be placed (cloned, copied, or
```
Comment (fryeguy52) [inline cmake/tribits/examples/TribitsExampleProjectAddons/README]: This TriBITS repository is designed ... 
(there is a bonus "is")

**Pull request merged**